webui/react/src/hooks/usePermissions.ts

```diff
@@ -399,6 +404,25 @@ const canModifyModelVersion = (
   return !rbacEnabled || permitted.has(V1PermissionType.EDITMODELREGISTRY);
 };
 
+// Template actions
+const canCreateTemplate = ({ rbacEnabled, userRoles }: RbacOptsProps): boolean => {
+  return (
+    !rbacEnabled ||
+    (!!userRoles &&
+      !!userRoles.find(
+        (r) => !!r.permissions.find((p) => p.id === V1PermissionType.CREATETEMPLATES),
+      ))
+  );
+};
```

Comment:
**noop:** this looks good, just a minor gotcha that I have seen in the past where the returned was a number or a string and in the case of `!!0` and `!!''` (empty string), they evaluate to be false and sometimes that is unexpected. Here it looks like they will be objects so the double bang works great.